src/Symfony/Component/FeatureFlag/FeatureChecker.php

```diff
@@ -0,0 +1,37 @@
+<?php
+
+/*
+ * This file is part of the Symfony package.
+ *
+ * (c) Fabien Potencier <fabien@symfony.com>
+ *
+ * For the full copyright and license information, please view the LICENSE
+ * file that was distributed with this source code.
+ */
+
+namespace Symfony\Component\FeatureFlag;
+
+final class FeatureChecker implements FeatureCheckerInterface
+{
+    private array $cache = [];
+
+    public function __construct(
+        private readonly FeatureRegistry $featureRegistry,
+        private readonly mixed $default,
+    ) {
+    }
+
+    public function isEnabled(string $featureName, mixed $expectedValue = true): bool
+    {
+        return $this->getValue($featureName) === $expectedValue;
+    }
+
+    public function getValue(string $featureName): mixed
+    {
+        try {
+            return $this->cache[$featureName] ??= $this->featureRegistry->get($featureName)();
+        } catch (FeatureNotFoundException) {
```

Comment:
We can add a `has(string $featureName): bool` method to the `FeatureRegistryInterface` to keep a not-nullable return value for `get`. WDYT ?

Reply:
Could it implement `Psr\Container\ContainerInterface` if so?

Reply:
It could. Before applying this change, let's see if we need this method. If we throw an exception when a feature is not found, there will be no need for a such method.

Reply:
Done.